conda_smithy/variant_algebra.py

```diff
@@ -126,15 +126,41 @@ def op_variant_key_add(v1: dict, v2: dict):
 
     If an ordering reorders the primary key all the zip_keys referring to that primary key will also
     be reodered in the same manner.
+
+    additional_zip_keys can be specified to either create a new zip_key or
+    add additional keys to the zip_key keyset containing the primary key
     """
     primary_key = v2["__migrator"]["primary_key"]
+    additional_zip_keys = v2["__migrator"].get("additional_zip_keys", [])
+
+    newly_added_zip_keys = set()
+
+    result = v1.copy()
+
+    if additional_zip_keys:
+        for chunk in result.get("zip_keys", []):
+            zip_keyset = set(chunk)
+            if primary_key in zip_keyset:
+                # The primary is already part of some zip_key, add the additional keys
+                for additional_key in additional_zip_keys:
+                    if additional_key not in zip_keyset:
+                        chunk.append(additional_key)
+                        newly_added_zip_keys.add(additional_key)
+                break
+        else:
+            # The for loop didn't break thus the primary is not part of any zip_key,
+            # create a new one including the primary key
+            result.setdefault("zip_keys", []).append(
+                [primary_key] + additional_zip_keys
+            )
+            newly_added_zip_keys.update([primary_key] + additional_zip_keys)
+
```

Comment:
This should be moved to after line 162 as this should only be used for key_add operation.

Reply:
Doesn't `op_variant_key_add` only get used for key_add operations? 😅

Reply:
It should, but I'm not sure why Line 160 returns fast.

Reply:
I see